Sources_v3/StreamChat/Database/DTOs/UserDTO.swift

```diff
@@ -21,13 +21,35 @@ class UserDTO: NSManagedObject {
     
     @NSManaged var flaggedBy: CurrentUserDTO?
     
+    @NSManaged var payloadHash: Int64
+    
     /// Returns a fetch request for the dto with the provided `userId`.
     static func user(withID userId: UserId) -> NSFetchRequest<UserDTO> {
         let request = NSFetchRequest<UserDTO>(entityName: UserDTO.entityName)
         request.sortDescriptors = [NSSortDescriptor(keyPath: \UserDTO.id, ascending: false)]
         request.predicate = NSPredicate(format: "id == %@", userId)
         return request
     }
+    
+    override func willSave() {
+        super.willSave()
+        var hasher = Hasher()
+        hasher.combine(id)
+        hasher.combine(name)
+        hasher.combine(imageURL)
+        hasher.combine(userRoleRaw)
+        hasher.combine(userCreatedAt)
+        hasher.combine(userUpdatedAt)
+        hasher.combine(lastActivityAt)
+        hasher.combine(isOnline)
+        // hasher.combine(isInvisible)
+        hasher.combine(isBanned)
+        // hasher.combine(teams)
+        hasher.combine(extraData)
+        // Direct assignment cannot be used
+        // Since it'll generate new `willSave` calls causing recursion
+        assignIfDifferent(self, \.payloadHash, Int64(hasher.finalize()))
```

Comment:
I wanted to convey "this should be the same as payload's hash with the same values" :D Since logically `dto.hashValue` won't be the same as `payload.hashValue`, they have differing properties

Reply:
Interesting ... my mental model for this is that values `A` `B` `C` from DTO has the same hash as values `A` `B` `C` from the payload